src/irmin-pack/unix/ext.ml

```diff
@@ -351,6 +354,36 @@ module Maker (Config : Conf.S) = struct
                       Pack_key.v_direct ~offset ~length ~hash:entry.hash
                     in
                     Some key)
+
+          let create_one_commit_store t commit_key path =
+            let () = Io.mkdir path |> Errs.raise_if_error in
+            let* _launched =
+              start_exn ~use_auto_finalisation:false ~path_new_files:path t
+                commit_key
+            in
+            let generation = File_manager.generation t.fm + 1 in
+            let* () =
+              match t.running_gc with
+              | None -> assert false
+              | Some { gc; _ } -> Gc.finalise_without_swap gc
+            in
+            let () =
+              File_manager.create_one_commit_store t.config ~generation
+                ~new_store_root:path commit_key
+              |> Errs.raise_if_error
+            in
+            let branch_path = Irmin_pack.Layout.V4.branch ~root:path in
+            let* branch_store =
+              Branch.v ~fresh:true ~readonly:false branch_path
+            in
+            (* Gc worker always fsyncs prefix and mapping files. File_manager
+               fsyncs dict, control, suffix and index. Fsync branch file here. Then
+               fsync directory. *)
+            let use_fsync = Irmin_pack.Conf.use_fsync t.config in
+            let () = if use_fsync then Branch.fsync branch_store in
+            let* () = Branch.close branch_store in
+            let () = Io.fsync_dir path |> Errs.raise_if_error in
```

Comment:
For a future when file manager also manages the branch store (https://github.com/mirage/irmin/pull/2125#discussion_r1015589263), all this code could live in `File_manager.create_one_commit_store` which would be cleaner and only require a comment about GC handling prefix/mapping files.

Reply:
yes, agree, these sort of comment show that there is an issue in the structure of the code